pkg/cmd/operator/probeserver/serveprobes.go

```diff
@@ -0,0 +1,100 @@
+package probeserver
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"sync"
+
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	"github.com/scylladb/scylla-operator/pkg/signals"
+	"github.com/scylladb/scylla-operator/pkg/version"
+	"github.com/spf13/cobra"
+	apierrors "k8s.io/apimachinery/pkg/util/errors"
+	cliflag "k8s.io/component-base/cli/flag"
+	"k8s.io/klog/v2"
+)
+
+type ServeProbesOptions struct {
+	Address string
+	Port    uint16
+
+	handler http.Handler
+}
+
+func NewServeProbesOptions(streams genericclioptions.IOStreams, port uint16, handler http.Handler) *ServeProbesOptions {
+	return &ServeProbesOptions{
+		Address: "",
+		Port:    port,
+		handler: handler,
+	}
+}
+func (o *ServeProbesOptions) AddFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&o.Address, "address", "", o.Address, "Listen address for the server.")
+	cmd.Flags().Uint16VarP(&o.Port, "port", "", o.Port, "Port to use for the server.")
+}
+
+func (o *ServeProbesOptions) Validate(args []string) error {
+	var errs []error
+
+	return apierrors.NewAggregate(errs)
+}
+
+func (o *ServeProbesOptions) Complete(args []string) error {
+	return nil
+}
+
+func (o *ServeProbesOptions) Run(originalStreams genericclioptions.IOStreams, cmd *cobra.Command) (returnErr error) {
+	klog.Infof("%s version %s", cmd.Name(), version.Get())
+	cliflag.PrintFlags(cmd.Flags())
```

Comment:
ditto

Reply:
this one is even used by `ScyllaDBAPIStatusOptions` at the end of `RunCtx`